src/dynamic_boundary_conditions/sea_level_rise_data.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Oct 18 16:30:21 2022.
+
+@author: alo78
+         xander.cai@pg.canterbury.ac.nz
+
+Used to get the sea level rise predictions (several predictions options to choose from)
+for site closest to the chosen latitude & longitude (Anywhere on the NZ Coastline) point
+from the csv's downloaded from: https://searise.takiwa.co/.
+"""
+
+import pandas as pd
+import geopandas as gpd
+from haversine import haversine
+import os
+from shapely.geometry import Polygon
+from src.digitaltwin import setup_environment
+
+
+def gen_dataframe(file_path: str) -> pd.DataFrame:
+    """ read csv file from input directory."""
+    df_list = []
+    for filename in os.listdir(file_path):
+        if filename.endswith('.csv'):
+            try:
+                #  pyarrow is faster, need Pandas 1.4, released in January 2022
+                df = pd.read_csv(file_path + filename, engine='pyarrow', dtype={'siteId': int})
+                print('LOADED CSV file with pyarrow: ', filename)
+            except (Exception, ):
+                df = pd.read_csv(file_path + filename, dtype={'siteId': int})
```

Comment:
siteId needs to be in a different data type, e.g. str.

Reply:
The code doesn't run on northland_region_1 otherwise

Reply:
It was for checking the wrong site id. Will change it to string type and commit it in the next commit.